src/document/DocumentAnnotator.ts

```diff
@@ -93,43 +82,17 @@ export default class DocumentAnnotator extends BaseAnnotator {
         }
 
         const annotation = getAnnotation(this.store.getState(), annotationId);
-        const annotationPage = annotation?.target.location.value;
+        const annotationPage = annotation?.target.location.value ?? 1;
+        const annotationPageEl = this.getPage(annotationPage);
 
-        if (!annotation || !annotationPage) {
+        if (!annotation || !annotationPage || !annotationPageEl || !this.annotatedEl) {
             return;
         }
 
         if (isRegion(annotation)) {
-            this.scrollToLocation(annotationPage, centerRegion(annotation.target.shape));
-        }
-    }
-
-    scrollToLocation(page = 1, { x: offsetX, y: offsetY } = { x: 0, y: 0 }): void {
-        const pageEl = this.getPage(page);
-
-        if (!this.annotatedEl || !pageEl) {
-            return;
-        }
-
-        const canSmoothScroll = 'scrollBehavior' in this.annotatedEl.style;
-        const parentCenterX = Math.round(this.annotatedEl.clientWidth / 2);
-        const parentCenterY = Math.round(this.annotatedEl.clientHeight / 2);
-        const offsetCenterX = Math.round(pageEl.clientWidth * (offsetX / 100));
-        const offsetCenterY = Math.round(pageEl.clientHeight * (offsetY / 100));
-        const offsetScrollLeft = pageEl.offsetLeft - parentCenterX + offsetCenterX;
-        const offsetScrollTop = pageEl.offsetTop - parentCenterY + offsetCenterY;
-        const scrollLeft = Math.max(0, Math.min(offsetScrollLeft, this.annotatedEl.scrollWidth));
-        const scrollTop = Math.max(0, Math.min(offsetScrollTop, this.annotatedEl.scrollHeight));
-
-        if (canSmoothScroll && Math.abs(this.annotatedEl.scrollTop - scrollTop) < SCROLL_THRESHOLD) {
-            this.annotatedEl.scrollTo({
-                behavior: 'smooth',
-                left: scrollLeft,
-                top: scrollTop,
+            scrollToLocation(this.annotatedEl, annotationPageEl, {
+                offsets: centerRegion(annotation.target.shape),
```

Comment:
Can this be abstracted out more in a follow up?

Reply:
How would you suggest it to be further abstracted? We have to differentiate between annotation types at some level so we can handle their measurements properly.

Reply:
Wondering if it the base scrollToAnnotation could do more, meaning less implementation for new annotators. No problem for now!